types/utxo.go

```diff
@@ -0,0 +1,142 @@
+package types //might need to be adjusted
+
+import (
+	crypto "github.com/tendermint/go-crypto"
+	"errors"
+	//sdk "github.com/cosmos/cosmos-sdk/types"
+
+)
+
+// UTXO is a standard unspent transaction output
+// Has pubkey for authentication
+type UTXO interface {
+	// Decided to use address instead of public keys
+	// since each utxo newly created will be created from an
+	// Ethereum address
+	GetAddress() crypto.Address
+	SetAddress(crypto.Address) error // errors if already set
+
+	//Get and Set denomination of the utxo. Is uint64 appropriate type?
```

Comment:
in the SDK, their Coin struct is using int64 for the Amount field, but `uint64` seems to make more sense... not sure why they want to have negative coins.